flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -80,14 +85,16 @@ public FlinkCatalog(
       String catalogName,
       String defaultDatabase,
       String[] baseNamespace,
-      Catalog icebergCatalog,
+      CatalogLoader catalogLoader,
       boolean cacheEnabled) {
     super(catalogName, defaultDatabase);
-    this.originalCatalog = icebergCatalog;
-    this.icebergCatalog = cacheEnabled ? CachingCatalog.wrap(icebergCatalog) : icebergCatalog;
+    this.originalCatalog = catalogLoader.loadCatalog(
+        HadoopUtils.getHadoopConfiguration(GlobalConfiguration.loadConfiguration()));
+    this.catalogLoader = catalogLoader;
+    this.icebergCatalog = cacheEnabled ? CachingCatalog.wrap(originalCatalog) : originalCatalog;
     this.baseNamespace = baseNamespace;
-    if (icebergCatalog instanceof SupportsNamespaces) {
-      asNamespaceCatalog = (SupportsNamespaces) icebergCatalog;
```

Comment:
Nice catching,  the CachingCatalog should never be a `SupportsNamespaces` catalog.

Reply:
previous code is correct too, `icebergCatalog` is the catalog.

Reply:
I mean: if we enable the `cacheEnabled`, then the icebergCatalog is the wrapped `Catalog` which is a `CachingCatalog` instance  actually.    But seems `CachingCatalog` did not implement the interface `SupportsNamespaces`.

Reply:
OK, this is the difference between `icebergCatalog` and `this.icebergCatalog`.  `icebergCatalog` could be a `SupportsNamespaces` catalog,  while `this.icebergCatalog` won't be...